packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -1802,7 +1837,7 @@ class OpacityLayer extends OffsetLayer {
   @override
```

Comment:
why all the work to switch to using a double if the engine just takes an int?

Reply:
Well there is also: https://github.com/flutter/engine/pull/33035

Reply:
Yes, we need to be driving towards deeper components just from the perspective of round off errors on parameters that will be multiplied a lot during the rendering pipeline, and >8-bit color components on screens is already happening.